sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -193,6 +193,24 @@ object DateTimeUtils {
     millisToDays(date.getTime)
   }
 
+  /**
+   * Returns the number of days since epoch from java.util.Calendar
+   */
+  def fromJavaCalendarForDate(cal: Calendar): SQLDate = {
+    val ms = cal.getTimeInMillis
+    cal.getTimeZone match {
+      case null => millisToDays(ms)
+      case tz => millisToDays(ms, tz)
+    }
+  }
+
+  /**
+   * Returns SQLTimestamp from java.util.Calendar (microseconds since epoch)
```

Comment:
(Matching the comment of fromJavaCalendarForDate.)
nit: Returns the number of microseconds since epoch from java.util.Calendar.